### DISCUSSION TITLE
Test de método con callback en ejercicio FileSystem

### DISCUSSION THREAD

**Discussion by msaavedranunez** [dds-jv/foro#401] [Consultas de objetos] [answered]
Hola.
Para poder hacer el test en asyncRead() estoy mockeando mis callback. El Consumer<Buffer> unConsumerBuffer lo uso en mi método y el Consumer <Integer> lo necesita el lowLevelFileSystem. Leyendo otros discussion en el foro, logré construir los dos con Mock. Cuando uso el primero me funciona bien pero cuando cuando uso el Consumer<Integer> en el Verify, me rompe. ¿Qué no estoy viendo? Muchas gracias.

![image](https://github.com/dds-jv/foro/assets/130939679/99382065-02f5-4dcf-8e85-9813a01fd339)

**Comment by flbulgarelli** [comment, marked as answer]
¡Hola @msaavedranunez !

El problema es que estás mezclando en un mismo verify argumentos literales con matchers. Mira acá https://javadoc.io/doc/org.mockito/mockito-core/3.3.3/org/mockito/ArgumentMatchers.html 

 Concretamente:

> Warning:
If you are using argument matchers, all arguments have to be provided by matchers. E.g: (example shows verification but the same applies to stubbing)

Por eso motivo te dice que se registraron 3 matchers (los tres del medio) y no cinco (el del 42 y tu _consumer_). Te va a convenir utilizar `eq` y `same`.

¡Decime si se entendió!

**Reply by msaavedranunez** [reply]
Ah claro!! Si si, muchas gracias.
Para corregirlo usé `eq(42)` y `eq(unConsumerInteger)` pero me tiraba un nuevo error, no reconociendo mi unConsumerInteger mockeado como el tipo de argumento esperado. Probé con `same` y lo mismo. Lo terminé arreglando pasándole como parámetro un `ArgumentMatchers.any()` y así por fin pasó el test.
Muchas gracias nuevamente.
Saludos.
